contracts/ERC721A.sol

```diff
@@ -613,6 +613,22 @@ contract ERC721A is IERC721A {
         }
     }
 
+    /**
+     * @dev Returns whether `spender` is allowed to manage `tokenId`.
+     *
+     * Requirements:
+     *
+     * - `tokenId` must exist.
+     */
+    function _isApprovedOrOwner(address spender, uint256 tokenId) internal view virtual returns (bool) {
```

Comment:
Follow up to that, lines 651-655 do the same thing as this function - we're checking if a tokenId can be transferred by the msg sender. I'm almost of the mindset that we should just use this function instead of those lines so the code is easier to read at the cost of some extra gas. Either way, I think we need to reduce duplicate logic.

Reply:
I think it is ok to duplicate logic.

The Solidity compiler refuses to inline any function with branching.

Reply:
I'm talking about readability here - it's becoming more and more confusing to track what function does what.

Reply:
I think this extension is a good opportunity to reintroduce the `_isApprovedOrOwner` function too. 

Can be convenient for 3rd party extensions.

Reply:
I would rename this function to be `_isSpenderApprovedForToken` to make it clearer that you're acting on a tokenId. In addition to that, I don't like the term `spender` in ERC721A because this concept is tied closer to 4907. 

I still don't think we need in 721A because extensions can still call isApprovedForAll and getApproved if they really need this information.

Reply:
Oh ya... why did I miss this!